src/jit/lsra.cpp

```diff
@@ -3742,18 +3743,20 @@ void LinearScan::buildRefPositionsForNode(GenTree*                  tree,
 
     bool isSpecialPutArg = false;
 
-    assert(!tree->OperIsAssignment());
     if (tree->OperIsLocalStore())
     {
-        if (isCandidateLocalRef(tree))
+        GenTreeLclVarCommon* const store = tree->AsLclVarCommon();
+        assert((consume > 1) || (regType(store->gtOp1->TypeGet()) == regType(store->TypeGet())));
+
+        LclVarDsc* varDsc = &compiler->lvaTable[store->gtLclNum];
+        if (isCandidateVar(varDsc))
```

Comment:
Again, these changes seem to be unrelated to this change, and I'd rather not see them conflated.